repository/src/main/java/org/pentaho/platform/repository2/unified/fileio/RepositoryFileOutputStream.java

```diff
@@ -179,7 +181,18 @@ public void flush() throws IOException {
     // RepositoryFileWriter
     // but I couldn't figure out a clean way to do that. For now, charsetName is passed in here and we use it if
     // available.
-    final SimpleRepositoryFileData payload = new SimpleRepositoryFileData( bis, charsetName, mimeType );
+    final IRepositoryFileData payload;
+    Converter converter;
+
+    if ( "ktr".equalsIgnoreCase( ext ) ) {
+      converter = PentahoSystem.get( Converter.class, null, Collections.singletonMap( "name", "PDITransformationStreamConverter" ) );
```

Comment:
Also, a unit or integration test would be nice. Not sure if it is possible, but would be nice.